homeassistant/components/template/coordinator.py

```diff
@@ -92,15 +99,33 @@ async def _attach_triggers(self, start_event=None) -> None:
         )
 
     async def _handle_triggered_with_script(self, run_variables, context=None):
+        if not self._check_condition(run_variables, context):
+            return
         # Create a context referring to the trigger context.
         trigger_context_id = None if context is None else context.id
         script_context = Context(parent_id=trigger_context_id)
         if script_result := await self._script.async_run(run_variables, script_context):
             run_variables = script_result.variables
-        self._handle_triggered(run_variables, context)
+        self._execute_update(run_variables, context)
+
+    async def _handle_triggered(self, run_variables, context=None):
+        if not self._check_condition(run_variables, context):
+            return
+        self._execute_update(run_variables, context)
+
+    def _check_condition(self, run_variables, context=None) -> bool | None:
```

Comment:
What does `None` mean as the return value?

Reply:
`None` is returned by condition checkers when the condition is disabled.

https://github.com/chammp/homeassistant-core/blob/d2991ae2491e76930870ebf8108a91e7ad0ba06f/homeassistant/helpers/condition.py#L119

https://github.com/chammp/homeassistant-core/blob/d2991ae2491e76930870ebf8108a91e7ad0ba06f/homeassistant/helpers/condition.py#L242-L249

The function returned by `async_conditions_from_config` does not return None though, maybe it should not use the `ConditionCheckerType`